crates/admin/src/cluster_controller/service.rs

```diff
@@ -225,18 +226,33 @@ impl ClusterControllerHandle {
     pub async fn create_partition_snapshot(
         &self,
         partition_id: PartitionId,
-    ) -> Result<Result<Snapshot, anyhow::Error>, ShutdownError> {
+        min_target_lsn: Option<Lsn>,
+        trim_log: Option<bool>,
+    ) -> Result<anyhow::Result<Snapshot>, ShutdownError> {
         let (response_tx, response_rx) = oneshot::channel();
 
         let _ = self
             .tx
             .send(ClusterControllerCommand::CreateSnapshot {
                 partition_id,
+                min_target_lsn,
                 response_tx,
             })
             .await;
 
-        response_rx.await.map_err(|_| ShutdownError)
+        let create_snapshot_response = response_rx.await.map_err(|_| ShutdownError)?;
+
+        if let (Ok(snapshot), Some(true)) = (&create_snapshot_response, trim_log) {
+            // todo(pavel): apply the safety logic from cluster auto-trim
+            if let Err(trim_error) = self
+                .trim_log(LogId::from(partition_id), snapshot.min_applied_lsn)
```

Comment:
Can we resolve the log-id from partition-table? This could be in a follow-up PR along with other occurrences like this.